src/graph/creation.rs

```diff
@@ -34,6 +34,8 @@ where
         };
         cg.add_connections(connections)?;
 
+        validate(&cg)?;
```

Comment:
done, I think it was because I thought there would be too many validation methods.  But you're right, as long as the individual validation methods are elsewhere, this one can be under `ComponentGraph`.